controller/resource.php

```diff
@@ -0,0 +1,45 @@
+<?php
+/**
+ * Nooku Framework - http://nooku.org/framework
+ *
+ * @copyright   Copyright (C) 2011 - 2015 Johan Janssens and Timble CVBA. (http://www.timble.net)
+ * @license     GNU GPLv3 <http://www.gnu.org/licenses/gpl.html>
+ * @link        http://github.com/joomlatools/joomlatools-framework-activities for the canonical source repository
+ */
+
+/**
+ * Resource Controller.
+ *
+ * @author  Arunas Mazeika <https://github.com/amazeika>
+ * @package Koowa\Component\Activities
+ */
+class ComActivitiesControllerResource extends KControllerModel
+{
+    protected function _initialize(KObjectConfig $config)
+    {
+        if ($this->getIdentifier()->getPackage() != 'activities')
```

Comment:
The cool part of dynamically aliasing like this is that it only happens if the classes aren't present on the main component and there's zero code to be added. It would basically work on its own and the way it's intended to with no intervention or even understanding of the developer.

This can also be done while bootstrapping yes, but it's way less friendly and requires good understanding on how to extend components. We do the exact same thing with the activity controller and this works really great. I would keep this.